packages/launch/src/hooks/use-title.ts

```diff
@@ -13,13 +15,40 @@ import LaunchContext from '../context';
 
 export function useTitle() {
 	const { siteId } = useContext( LaunchContext );
-	const title = useSelect( ( select ) => select( SITE_STORE ).getSiteTitle( siteId ) );
-	const launchSiteTitle = useSelect( ( select ) => select( LAUNCH_STORE ) ).getSiteTitle();
-	const setLaunchSiteTitle = useDispatch( LAUNCH_STORE ).setSiteTitle;
+
+	const locale = useLocale();
+
+	const existingSiteTitle = useSelect( ( select ) => select( SITE_STORE ).getSiteTitle( siteId ), [
+		siteId,
+	] );
+	const launchSiteTitle = useSelect( ( select ) => select( LAUNCH_STORE ).getSiteTitle(), [] );
+
+	const updateTitle = useDispatch( LAUNCH_STORE ).setSiteTitle;
 	const saveSiteTitle = useDispatch( SITE_STORE ).saveSiteTitle;
+
 	const [ debouncedSaveSiteTitle ] = useDebouncedCallback( saveSiteTitle, 1000 );
 
 	const noLaunchSiteTitle = typeof launchSiteTitle === 'undefined';
+	const title = noLaunchSiteTitle ? existingSiteTitle : launchSiteTitle;
+
+	const DEFAULT_SITE_NAME = __( 'Site Title', __i18n_text_domain__ );
+	let isDefaultTitle = title === DEFAULT_SITE_NAME;
+	let isValidTitle = title !== '' && ! isDefaultTitle;
```

Comment:
```suggestion
	let isValidTitle = typeof title !== 'undefined' && title !== '' && ! isDefaultTitle;
```

or, less explicit but shorter:

```suggestion
	let isValidTitle = Boolean(title) && ! isDefaultTitle;
```

See https://github.com/Automattic/wp-calypso/pull/50311#discussion_r580179613 for context